app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -116,6 +116,11 @@ def update_auth_header
   private
 
   def refresh_headers
+    # At this point restore the attributes so we can obtain the lock
+    # otherwise an exception is thrown while trying to obtain the lock
+    # with unsaved data on the resource.
+    @resource.restore_attributes
```

Comment:
This doesn't exist in Mongo 😞

Reply:
:(

Reply:
Maybe leave the previous behavior for Mongo and this for rails. But split in 2 PRs so is easily done and we avoid breaking changes.

Reply:
Hi, I apologize for my late reply. How can I check if the database is going to be Mongo?

Reply:
`if const_defined?('Mongoid')`